### ISSUE TITLE
Response content-type as `text/html` with content in `json` cause content area display empty

### ISSUE THREAD

**Issue by BLumia** [liyasthomas/postwoman#869]
**Describe the bug**
When the returend content-type is `text/plain` or `text/html` but the actual content is a json, the preview shows empty.

**To Reproduce**

I did a quick and dirty server side demo implementation in go:

``` go
package main

import "fmt"
import "log"
import "strings"
import "strconv"
import "net/http"

func FileServer(response http.ResponseWriter, request *http.Request) {
	path := request.URL.Path

	var contentType string;
	responseType := path[strings.LastIndex(path, "/"):]

	switch responseType {
	case "/json":
        contentType = "application/json";
	case "/plain":
        contentType = "text/plain";
	default:
        contentType = "text/html";
	}
	
	response.Header().Set("content-type", contentType)

    fmt.Fprintf(response, "{\"status\": 200, \"content-type\": \"%s\"}", contentType)
}

func Controller(response http.ResponseWriter, request *http.Request) {
	FileServer(response, request)
}

func main() {
	PortNumber := 4000

	fmt.Println("Now listening port: " + strconv.Itoa(PortNumber))

	http.HandleFunc("/", Controller)
	err := http.ListenAndServe(":"+strconv.Itoa(PortNumber), nil)
	if err != nil {
		log.Fatal(err)
	}
}
```

Do compile it (`go compile main.go`) and run it, then request the following address:

 - `http://127.0.0.1:4000/` : response "text/html" as content-type, preview is empty.
 - `http://127.0.0.1:4000/plain` : response "text/plain" as content-type, preview is empty.
 - `http://127.0.0.1:4000/json` : response "application/json" as content-type, preview is expected.

Also, when the preview is empty, try toggle "Preview HTML" button, then it will show a `[object Object]`, but "Download File" icon button can download the actual response content.

**Expected behavior**

When response content-type is html or plain, the preview area should able to at least show the raw content instead of shows empty.

**Screenshots**

Following image is what it shows when request `http://127.0.0.1:4000/` with the above demo server program running.

![response content area shows empty](https://user-images.githubusercontent.com/10095765/81766804-7053b600-9509-11ea-879b-4ce9f5338a52.png)
![response content area shows object Object](https://user-images.githubusercontent.com/10095765/81766831-7fd2ff00-9509-11ea-87fb-8f6d0c37c33d.png)

**Desktop (please complete the following information):**
 - OS: Windows 10 (I guess it doesn't matter what OS I am using)
 - Browser chrome
 - Version 81.0.4044.138

Let me know if you need any additional context to resolve this issue. Hope that helps.

**Comment by liyasthomas** [comment]
> Response content-type as `text/html` with content in `json`
Is this an actual real-life case or just for testing? I've never seen such an endpoint.

**Comment by BLumia** [comment]
> > Response content-type as `text/html` with content in `json`
> > Is this an actual real-life case or just for testing? I've never seen such an endpoint.

It's in real life. I just got this issue today with a endpoint return `text/html` with a json content. Since that service is not expode to public usage so I cannot link the real API here, sorry about that.

I agree that endpoint implementation might not do the best practice, but still, I do think the content area in postwoman should at least display the raw content if the content-type is `text/html` or `text/plain`, instead of display a `[object Object]`...

**Comment by liyasthomas** [comment]
Issue has been acknowledged. I'm working on a fix.

**Comment by AndrewBastin** [comment]
@liyasthomas are you working on this ?

Else I will take it up ^_^

**Comment by liyasthomas** [comment]
@AndrewBastin Actually I couldn't find why this behaviour happened. The contents passed to the editor seens to be receiving an `object` type.

**Comment by AndrewBastin** [comment]
I am working on it.

Seems like it is a weird Ace Editor bug.

**Comment by AndrewBastin** [comment]
This bug is caused due to the default behaviour of axios to take string responses and try to parse them to JSON objects even if the content type is not for a JSON response.

So, when a valid returned eventhough it is set to `text/plain`, it will try parsing it and get the JSON object. Then Postwoman can't figure out what to do with JSON object, causing the issue.

This behaviour is overridden on the update to the AxiosStrategy in my PR (#881).

But, we have to handle it in all the places where we use Axios.

The browser extension will receive the fix to this in the coming days in a 0.7 update.
Meanwhile, if the proxy uses axios, that should be handled as well @NBTX .

**Comment by AndrewBastin** [comment]
@NBTX, I haven't personally tested this with the proxy, but is this an issue on the proxy ?

@liyasthomas ?
